theseus/optimizer/manifold_gaussian.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from itertools import count
+from typing import Optional, Sequence
+
+import torch
+
+from theseus.geometry import Manifold
+
+
+class ManifoldGaussian:
+    _ids = count(0)
+
+    def __init__(
+        self,
+        mean: Sequence[Manifold],
+        precision: Optional[torch.Tensor] = None,
+        name: Optional[str] = None,
+    ):
+        self._id = next(ManifoldGaussian._ids)
+        if name:
+            self.name = name
+        else:
+            self.name = f"{self.__class__.__name__}__{self._id}"
+
+        dof = 0
+        for v in mean:
+            dof += v.dof()
+        self._dof = dof
+
+        self.mean = mean
+        self.precision = torch.zeros(mean[0].shape[0], self.dof, self.dof).to(
```

Comment:
Since precision is an optional argument to init you want a 'if none' check here before assigning zeros, and if something is passed then you can add the check for dtype/device Luis suggested as well as check for the shape to be correct (and if not then raise value error with a message).